### PR TITLE
Model fitting

### PR DESCRIPTION
**Pull request by b-remy**
Pull request to add model fitting functions taking as input a galaxy image and returning its flux, ellipticities and radius.

It still remains to add tests.

### CODE REVIEW

**Comment by b-remy** [comment]
I added two tests checking that we can estimate flux and shapes (e1, e2) on a galsim example with errors `1e-1`   for flux and `1e-2` for shapes.

Fell free to ask for other tests or any change :)

File: autometacal/tests/test_model_fitting.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+# Draw the image with Galsim
+image_galsim = gal.drawImage(nx=N,ny=N,scale=pixel_scale,method='no_pixel').array
+
+# Estimate flux and shape (e1,e2) with model fitting
+##############################
+_, e, flux, _ = fitting.fit_multivariate_gaussian(image_galsim,
+                                                  pixel_scale,
+                                                  update_params={'lr':30.})
+
+def test_fitting_flux():
```
Comment (EiffL) [inline autometacal/tests/test_model_fitting.py:40]: This looks great :-) but usually we encapsulate each test as a self contained function. And you can have several asserts in the same function.
Maybe the galsim code doesn't absolutely need to be in the test function, but it's better if the function that we are testing is called inside the test function.
So here you could have a `test_fitting()` test that runs all of the galsim code and have 2 asserts to check the shape and flux

**Review by EiffL — CHANGES_REQUESTED** [review]
This looks good :-) Just a small comment on the tests

File: autometacal/tests/test_model_fitting.py (hunk 1 above)
Comment (b-remy) [reply autometacal/tests/test_model_fitting.py]: Ok, I was not sure that I could put several asserts in the same test function. I checked that it works and updated.

File: autometacal/tests/test_model_fitting.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  gal = galsim.Exponential(flux=gal_flux, scale_radius=gal_r0)
+  # To make sure that GalSim is not cheating, i.e. using the analytic formula of the light profile
+  # when computing the affine transformation, it might be a good idea to instantiate the image as
+  # an interpolated image.
+  # We also make sure GalSim is using the same kind of interpolation as us (bilinear for TF)
+  reference_image = gal.drawImage(nx=N,ny=N, scale=pixel_scale)
+  gal = galsim.InterpolatedImage(reference_image,
+                                 x_interpolant='linear')
+
```
Comment (aguinot) [inline autometacal/tests/test_model_fitting.py:34]: In that case you don't need to go through the interpolation of the image and apply an affine transform. You can use the analytic formula which is closer to what happen in reality. Basically you can do:
```
gal = galsim.Exp()
gal = gal.shear()
image_galsim = gal.drawImage()
```
In that case you would use method='auto' for the drawing (which is the default).

File: autometacal/fitting.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+import numpy as np
+
+import galflow as gf
+
+def fit_multivariate_gaussian(image, pixel_scale, update_params=None):
+  """
+  Estimate galaxy parameters by fitting a multivariate gaussian
+
+  Args:
+      image: galaxy image to fit
```
[the remark this answers is not in the document]
Comment (b-remy) [reply autometacal/fitting.py]: I checked and commented that both numpy and TF inputs work

File: autometacal/tests/test_model_fitting.py (hunk 2 above)
Comment (b-remy) [reply autometacal/tests/test_model_fitting.py:30]: Alright, done!

**Review by EiffL — APPROVED** [review]
LGTM

**Pull request merged**